From 8aa5bdcc1683fcfe65429a8a2326972f90d5f0e9 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?R=C3=A9mi=20Bernon?= <rbernon@codeweavers.com>
Date: Tue, 9 Dec 2025 11:05:10 +0100
Subject: [PATCH 17/33] win32u: Remove unnecessary drawable flush in
 context_sync_drawables.

This fixes sluggish rendering of Steam and other CEF based applications,
as wined3d calls wglMakeCurrent with possibly already active surface
every time it tries to activate a context.
---
 dlls/win32u/opengl.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/dlls/win32u/opengl.c b/dlls/win32u/opengl.c
index 398a984e23e..db659ba52f9 100644
--- a/dlls/win32u/opengl.c
+++ b/dlls/win32u/opengl.c
@@ -1692,12 +1692,13 @@ static BOOL context_sync_drawables( struct wgl_context *context, HDC draw_hdc, H
 
         opengl_drawable_set_context( new_read, context );
         if (new_read != new_draw) opengl_drawable_set_context( new_draw, context );
+
+        opengl_drawable_flush( new_read, new_read->interval, 0 );
+        opengl_drawable_flush( new_draw, new_draw->interval, 0 );
     }
 
     if (ret)
     {
-        opengl_drawable_flush( new_read, new_read->interval, 0 );
-        opengl_drawable_flush( new_draw, new_draw->interval, 0 );
         /* update the current window drawable to the last used draw surface */
         if (new_draw->client) set_window_opengl_drawable( new_draw->client->hwnd, new_draw, TRUE );
         context_exchange_drawables( context, &new_draw, &new_read );
-- 
2.52.0

